src/commands/stepfunctions/uninstrument.ts

```diff
@@ -0,0 +1,155 @@
+      ],
+    ],
+  })
+
+  private dryRun = false
+  private stepFunctionArns: string[] = []
+
+  public async execute() {
+    let validationError = false
+    let hasChanges = false
+
+    // remove duplicate step function arns
+    const stepFunctionArns = [...new Set(this.stepFunctionArns)]
+
+    if (stepFunctionArns.length === 0) {
+      this.context.stdout.write(`[Error] must specify at least one \`--step-function\`\n`)
+      validationError = true
+    }
+
+    for (const stepFunctionArn of stepFunctionArns) {
+      if (!isValidArn(stepFunctionArn)) {
+        this.context.stdout.write(`[Error] invalid arn format for \`--step-function\` ${stepFunctionArn}\n`)
+        validationError = true
+      }
+    }
+
+    if (validationError) {
+      return 1
+    }
+
+    // loop over step functions passed as parameters and generate a list of requests to make to AWS for each step function
+    for (const stepFunctionArn of stepFunctionArns) {
+      // use region from the step function arn to make requests to AWS
+      const arnObject = parseArn(stepFunctionArn)
+      const region = arnObject.region
+      const cloudWatchLogsClient = new CloudWatchLogsClient({region})
+      const stepFunctionsClient = new SFNClient({region})
+
+      let describeStateMachineCommandOutput
+      try {
+        describeStateMachineCommandOutput = await describeStateMachine(stepFunctionsClient, stepFunctionArn)
+      } catch (err) {
+        if (err instanceof Error) {
+          this.context.stdout.write(`\n[Error] ${err.message}. Unable to fetch Step Function ${stepFunctionArn}\n`)
+        }
+
+        return 1
+      }
+
+      const logGroupArn = getStepFunctionLogGroupArn(describeStateMachineCommandOutput)
+      if (logGroupArn === undefined) {
+        this.context.stdout.write('\n[Error] Unable to get Log Group arn from Step Function logging configuration\n')
+
+        return 1
+      }
+      const logGroupName = parseArn(logGroupArn).resourceName
+
+      // delete subscription filters that are created by Datadog-CI
+      let describeSubscriptionFiltersResponse: DescribeSubscriptionFiltersCommandOutput | undefined
+      try {
+        describeSubscriptionFiltersResponse = await describeSubscriptionFilters(cloudWatchLogsClient, logGroupName)
+      } catch (err) {
+        if (err instanceof Error) {
+          this.context.stdout.write(
+            `\n[Error] ${err.message}. Unable to fetch Subscription Filter to delete for Log Group ${logGroupName}\n`
+          )
+        }
+
+        return 1
+      }
+      const subscriptionFilters =
+        describeSubscriptionFiltersResponse.subscriptionFilters?.filter((subscriptionFilter) =>
+          subscriptionFilter.filterName?.includes(DD_CI_IDENTIFING_STRING)
+        ) ?? []
+
+      for (const subscriptionFilter of subscriptionFilters) {
+        if (typeof subscriptionFilter.filterName === 'string') {
```

Comment:
Can `filterName` be of another type?
I'm curious why this specific line.

Reply:
It could be `undefined` and linter would complain if we remove the check.
```
export interface SubscriptionFilter {
    filterName?: string;
   ...
}
```